src/OpenApi/test/Microsoft.AspNetCore.OpenApi.Tests/Services/OpenApiSchemaService/OpenApiSchemaService.ParameterSchemas.cs

```diff
@@ -722,4 +722,93 @@ public static bool TryParse(string value, out Student result)
             return true;
         }
     }
+
+    // Regression test for https://github.com/dotnet/aspnetcore/issues/62023
+    // Testing that the array parsing in our OpenApiJsonSchema works
+    [Fact]
+    public async Task CustomConverterThatOutputsArrayWithDefaultValue()
+    {
+        // Arrange
+        var serviceCollection = new ServiceCollection();
+        serviceCollection.ConfigureHttpJsonOptions(options =>
+        {
+            options.SerializerOptions.Converters.Add(new EnumArrayTypeConverter());
+        });
+        var builder = CreateBuilder(serviceCollection);
+
+        // Act
+        builder.MapPost("/api", (EnumArrayType e = EnumArrayType.None) => { });
+
+        // Assert
+        await VerifyOpenApiDocument(builder, document =>
+        {
+            var operation = document.Paths["/api"].Operations[OperationType.Post];
+            var param = Assert.Single(operation.Parameters);
+            Assert.NotNull(param.Schema);
+            Assert.IsType<OpenApiArray>(param.Schema.Default);
+            // Type is null, it's up to the user to configure this via a custom schema
+            // transformer for types with a converter.
+            Assert.Null(param.Schema.Type);
+        });
+    }
+
+    [Fact]
+    public async Task CustomConverterThatOutputsObjectWithDefaultValue()
+    {
+        // Arrange
+        var serviceCollection = new ServiceCollection();
+        serviceCollection.ConfigureHttpJsonOptions(options =>
+        {
+            options.SerializerOptions.Converters.Add(new EnumObjectTypeConverter());
+        });
+        var builder = CreateBuilder(serviceCollection);
+
+        // Act
+        builder.MapPost("/api", (EnumArrayType e = EnumArrayType.None) => { });
+
+        // Assert
+        await VerifyOpenApiDocument(builder, document =>
+        {
+            var operation = document.Paths["/api"].Operations[OperationType.Post];
+            var param = Assert.Single(operation.Parameters);
+            Assert.NotNull(param.Schema);
+            Assert.IsType<OpenApiObject>(param.Schema.Default);
+            // Type is null, it's up to the user to configure this via a custom schema
+            // transformer for types with a converter.
+            Assert.Null(param.Schema.Type);
+        });
+    }
+
+    public enum EnumArrayType
+    {
+        None = 1
+    }
+
+    public class EnumArrayTypeConverter : JsonConverter<EnumArrayType>
```

Comment:
The `EnumArrayTypeConverter` and `EnumObjectTypeConverter` classes has been used only in the tests. Can we not make those classes as `internal`?

```suggestion
    internal class EnumArrayTypeConverter : JsonConverter<EnumArrayType>
```